components/automate-cli/pkg/reportgenerator/usagegenerator.go

```diff
@@ -251,3 +263,257 @@ func queryElasticSearchNodeReport(client *elastic.Client, startTime time.Time, e
+	rangeQuery := elastic.NewRangeQuery("end_time").
+		Format("yyyy-MM-dd||yyyy-MM-dd-HH:mm:ss||yyyy-MM-dd'T'HH:mm:ssZ")
+	rangeQuery.Gte(startTime)
+	rangeQuery.Lte(endTime)
+
+	aggr := elastic.NewCardinalityAggregation().Field("node_uuid")
+	searchService := client.Search().
+		Query(rangeQuery).
+		Index("comp-7-r-*").
+		Size(100).
+		Aggregation("nodes_count", aggr)
+
+	searchResult, err := searchService.Do(context.Background())
+	if err != nil {
+		fmt.Println("Error in query: ", err)
+		os.Exit(1)
+	}
+
+	metric, ok := searchResult.Aggregations.ValueCount("nodes_count")
+	return metric, ok
+}
+
+func queryElasticSearchComplianceNodeReport(client *elastic.Client, startTime time.Time, endTime time.Time) {
+	filename := fmt.Sprintf("complianceinfo_%s_%s.csv", startTime.Format("2006-01-02"), endTime.Format("2006-01-02"))
+	f, err := os.OpenFile(filename, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0600)
+	if err != nil {
+		fmt.Println("Failed to open Compliance scan detail report")
+		os.Exit(1)
+	}
+	defer f.Close()
+
+	w := gocsv.DefaultCSVWriter(f)
+
+	var sourceField = []string{
+		"node_uuid",
+		"version",
+		"node_name",
+		"environment",
+		"platform.name",
+		"controls_sums.total",
+		"controls_sums.passed.total",
+		"controls_sums.skipped.total",
+		"controls_sums.failed.total",
+		"profiles.name",
+		"profiles.title",
+		"profiles.version",
+		"end_time",
+	}
+
+	t := endTime.Add(24 * time.Hour)
+
+	if t.Before(startTime) {
+		t = startTime
+	}
+	header := true
+	for {
+		rangeQuery := elastic.NewRangeQuery("end_time").
+			Format("yyyy-MM-dd||yyyy-MM-dd-HH:mm:ss||yyyy-MM-dd'T'HH:mm:ssZ")
+		rangeQuery.Gte(t)
+		rangeQuery.Lte(endTime)
+
+		fetchSource := elastic.NewFetchSourceContext(true).Include(sourceField...)
+
+		searchService := client.Search().
+			Query(rangeQuery).
+			Index("comp-7-r-*").
+			Size(10000).
+			FetchSourceContext(fetchSource)
+
+		searchResult, err := searchService.Do(context.Background())
+		if err != nil {
+			fmt.Println("Error in query: ", err)
+			os.Exit(1)
+		}
+
+		if searchResult.Hits.TotalHits > 0 {
+			if header == true {
+				headers := []string{"Node_ID", "Version", "Node_Name", "Environment", "End_Time", "Platform__Name", "Controls_Sums__Total", "Controls_Sums__Passed", "Controls_Sums__Skipped", "Controls_Sums__Failed", "Profiles__Count"}
+				if err := w.Write(headers); err != nil {
+					fmt.Println("Error while writing csv: ", err)
+				}
+				w.Flush()
+				if err := w.Error(); err != nil {
+					fmt.Println("Error while writing csv: ", err)
+				}
+			}
+
+			for _, hit := range searchResult.Hits.Hits {
+
+				var payload []byte
+
+				payload = []byte("[")
+				comma := []byte(",")
+				payload = append(payload, *hit.Source...)
+				payload = append(payload, comma...)
+				payload = payload[:len(payload)-1]
+				payload = append(payload, []byte("]")...)
+				var s ComplianceInfo
```

Comment:
var 's' can be declared outside for loop since it is anyway being assigned inside!